### PR TITLE
Feature/1185/promo code edit page

### PR DESCRIPTION
**Pull request by tarasSanotskyi**
## Description

Create promo code edit page

#### Screenshots

image
### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅ All tests passed locally
- [x] ✨ My changes working with up-to-date front-end and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/pages/promo-code/promo-code-edit/promo-code-edit.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import LoadingBar from '../../../components/loading-bar';
+import PromoCodeForm from '../promo-code-form/promo-code-form';
+
+function PromoCodeEdit() {
+  const { id } = useParams();
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const token = getFromLocalStorage(LOCAL_STORAGE.AUTH_ACCESS_TOKEN);
```
Comment (maarekua) [inline src/pages/promo-code/promo-code-edit/promo-code-edit.js:27]: Please, provide the token in general Apollo provider configuration.

File: src/pages/promo-code/promo-code-edit/promo-code-edit.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    onCompleted: onCompletedHandler,
+    context: {
+      headers: {
+        token
+      }
+    }
+  });
+
+  const pathToPromoCodesPage = config.routes.pathToPromoCodes;
+  const promoCode = data?.getPromoCodeById;
```
Comment (maarekua) [inline src/pages/promo-code/promo-code-edit/promo-code-edit.js:53]: It is better not to try getting data from properties that at this point in time could be undefined.

File: src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import { useMutation, useQuery } from '@apollo/client';
+
+import PromoCodeEdit from '../promo-code-edit';
+import { mocks } from './promo-code-edit.variables';
+
+jest.mock('react-redux');
+const dispatch = jest.fn();
+useDispatch.mockImplementation(() => dispatch);
+
+jest.mock('@apollo/client');
```
Comment (maarekua) [inline src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js:14]: We should not mock apollo/client.

https://www.apollographql.com/docs/react/development-testing/testing/

File: src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  it('test Loader in PromoCodeEdit component', () => {
+    const loader = screen.queryByRole(/'progressbar'/i);
+
+    expect(loader).toBeDefined();
+  });
+});
+
+describe('edit loading', () => {
+  beforeEach(() => {
+    useQuery.mockImplementation(() => ({
```
Comment (maarekua) [inline src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js:44]: Remove all the useQuery mocks.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/promo-code/promo-code-edit/promo-code-edit.js (hunk 1 above)
Comment (tarasSanotskyi) [reply src/pages/promo-code/promo-code-edit/promo-code-edit.js]: done

File: src/pages/promo-code/promo-code-edit/promo-code-edit.js (hunk 2 above)
Comment (tarasSanotskyi) [reply src/pages/promo-code/promo-code-edit/promo-code-edit.js]: done

File: src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js (hunk 3 above)
Comment (tarasSanotskyi) [reply src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js]: done

File: src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js (hunk 4 above)
Comment (tarasSanotskyi) [reply src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js]: done

File: src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <PromoCodeEdit />
+        </Router>
+      </MockedProvider>
+    );
+  });
+
+  it('should change promocode code name', async () => {
+    await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (SirAndrii) [inline src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js:40]: this line duplicates in each test, maybe move it to beforeEach Block?

Comment (tarasSanotskyi) [reply src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js]: done

**Review by bohdanst1914 — APPROVED** [review]

**Pull request merged**